Sources/CodeEditSourceEditor/Highlighting/StyledRangeContainer/StyledRangeStore/StyledRangeStore+StyledRun.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  StyledRangeStore+StyledRun.swift
+//  CodeEditSourceEditor
+//
+//  Created by Khan Winter on 10/25/24
+
+import _RopeModule
+
+extension StyledRangeStore {
+    struct StyledRun {
+        var length: Int
+        let capture: CaptureName?
+        let modifiers: CaptureModifierSet
+
+        static func empty(length: Int) -> Self {
+            StyledRun(length: length, capture: nil, modifiers: [])
+        }
+
+        /// Compare two styled ranges by their stored styles.
+        /// - Parameter other: The range to compare to.
+        /// - Returns: The result of the comparison.
+        func styleCompare(_ other: Self) -> Bool {
+            capture == other.capture && modifiers == other.modifiers
+        }
+    }
+}
+
+extension StyledRangeStore.StyledRun: RopeElement {
+    typealias Index = Int
+
+    var summary: Summary { Summary(length: length) }
+
+    @inlinable
+    var isEmpty: Bool { length == 0 }
+
+    @inlinable
+    var isUndersized: Bool { false } // Never undersized, pseudo-container
+
+    func invariantCheck() {}
+
+    mutating func rebalance(nextNeighbor right: inout Self) -> Bool {
+        // Never undersized
+        fatalError("Unimplemented")
+    }
+
+    mutating func rebalance(prevNeighbor left: inout Self) -> Bool {
+        // Never undersized
+        fatalError("Unimplemented")
+    }
```

Comment:
Are these methods never called, i.e don't we need them?

Reply:
They're required by the ``RopeElement`` protocol, but because we always return `false` for `isUndersized` they're never called. We're not actually storing a container in each rope element. Each element the length of the character 'container' it represents so it can't ever be undersized.